changelog.d/15224.feature

```diff
@@ -0,0 +1 @@
+Add `forget_rooms_on_leave` config option to automatically forget rooms when users leave them or are removed from them.
```

Comment:
Does this fix https://github.com/matrix-org/synapse/issues/4720 ?

We should update the issue description in that case

Reply:
No, this does not purge rooms from the database at all.

Reply:
This PR is adjacent, but I see it as more optional than anything. #4720 sounds like it's scoped to the case where all users explicitly choose to forget a room themselves. This PR forces users to forget rooms, making the proposed GC in #4720 trigger more often.

Reply:
🤷 Feels like a nice step to make the GC automatic (sane default)